text/0047-introduce-profile-context.md

```diff
@@ -0,0 +1,110 @@
+- Start Date: 2022-12-20
+- RFC Type: feature
+- RFC PR: https://github.com/getsentry/rfcs/pull/47
+- RFC Status: draft
+
+# Summary
+
+We want to introduce the concept of a profiling context, much like a tracing context, starting with
+transactions so we can easily relate a group of transactions back to the respective profile.
+
+# Motivation
+
+We already have a way to relate a group of profiles back to the respective transaction. And up to
+this point, we have been using this relationship to map single transaction at a time to the
+respective profile. This is now insufficient as we would like to be able to relate a group of
+transactions back to the respective profile. This is to enable various touch points into profiling
+in the existing performance product.
+
+The intended use cases include
+
+- fetching a list of transaction with the profile ID
+- filtering transactions to only those that have a profile
+
+# Background
+
+Profiles will be closely associated to transactions and creating this relation will enable us to
+build a richer integration between the Performance and Profiling products.
+
+Since the Profiling product will be initially tied to the Performance product, we want to ensure
+that there is a smooth integration through various touch points to allow users to navigate from
+Performance to Profiling and vice versa.
+
+# Options Considered
+
+## SDK
+
+### Option A: Add a new context type
+
+Introduce a key in the transaction payload under contexts called `profile` to include information
+related to the profile.
+
+```json
+{
+  ...,
+  "contexts": {
+    ...,
+    "profile": {
+      "profile_id": "<id>"
+    }
```

Comment:
Apart from `profile_id`, do we expect any other fields in this context in the future? If not, I would prefer option B or even a top-level `profile_id` field.

Reply:
We don't have any plans right now to add more fields to this context in the right now but that may change as we move towards a continuous profiling approach.